linodecli/plugins/k8s-alpha.py

```diff
@@ -0,0 +1,286 @@
+        return val
+    elif type(val) is str:
+        return '"{}"'.format(val)
+    else:
+        return None
+
+def gen_terraform_file(context, cluster_name, hashname):
+    tf_file_parts = []
+
+    for varname in tf_var_map.keys():
+        tf_file_parts.append("""variable "{tf_varname}" {{
+  default = {default}
+}}
+""".format(tf_varname=tf_var_map[varname]['name'],
+           default=quoted_string_or_bare_int(tf_var_map[varname]['default'])))
+    
+    tf_file_parts.append("""module "k8s" {{
+  source  = "git@github.com:linode/terraform-linode-k8s.git?ref=for-cli"
+
+  linode_token = "{token}"
+  
+  linode_group = "ka{hashname}-{cluster_name}"
+""".format(token=context.token,
+           cluster_name=cluster_name,
+           hashname=hashname,))
+
+    for varname in tf_var_map.keys():
+        tf_file_parts.append("""
+  {tf_varname} = "${{var.{tf_varname}}}"
+""".format(tf_varname=tf_var_map[varname]['name']))
+
+    tf_file_parts.append("}\n")
+
+    return ''.join(tf_file_parts)
+
+def gen_terraform_args(parsed):
+    args = []
+    for varname in tf_var_map.keys():
+        args = args + ['-var', "{}={}".format(tf_var_map[varname]['name'], getattr(parsed, varname))]
+    return args
+
+def call_or_exit(*args, **kwargs):
+    ret = spcall(*args, **kwargs)
+    if ret != 0:
+        sys.exit(ret)
+
+def replace_kubeconfig_user(terrapath, cluster_name):
+    """
+    If we leave the user as kubernetes-admin. Then the configs don't flatten properly.
+    All of them try to create creds for kubernetes-admin.
+    """
+    kubeconfig_fp = os.path.join(terrapath, "{}.conf".format(cluster_name))
+    kubeconfig = open(kubeconfig_fp).read()
+    kubeconfig = kubeconfig.replace('kubernetes-admin', cluster_name)
+
+    kubeconfig_new_fp = os.path.join(terrapath, "{}_new.conf".format(cluster_name))
+    kubeconfig_new = open(kubeconfig_new_fp, 'w')
+    kubeconfig_new.write(kubeconfig)
+
+    return kubeconfig_new_fp
+
+def safe_make_appdir(appname):
+    if sys.platform == 'win32':
+        appdir = os.path.join(os.environ['APPDATA'], appname)
+    else:
+        appdir = os.path.expanduser(os.path.join("~", "." + appname))
+
+    safe_mkdir(appdir)
+
+    return appdir
+
+def safe_mkdir(dirpath):
+    if not os.path.isdir(dirpath):
+        try:
+            os.mkdir(dirpath)
+        except:
+            print('Unable to create the directory: {}'.format(dirpath))
+            sys.exit(1)
+
+def get_hashname(name):
+    """
+    A cluster hashname is the first 9 characters of a SHA256 digest encoded in base64
+
+    This is used as a compact way to uniquely identify a cluster's Linode resources.
+    It's also stateless! If a user loses their terraform file and wishes to
+    delete a cluster they can still do so.
+    """
+    hashname = base64.b64encode(hashlib.sha256(name.encode('utf8')).digest())
+    hashname = hashname.decode('utf8')
+    hashname = hashname.replace(r'+', r'')
+    hashname = hashname.replace(r'/', r'')
```

Comment:
These work without being regex strings (i.e. `replace('+', '')`)

Reply:
I'll see myself out then 🤦‍♂️

Reply:
You're right that these don't need to be raw strings